training_main.py

```diff
@@ -0,0 +1,354 @@
+# %%
+
+# Importing generic python libraries
+import string
+import os
+import math
+import glob
+import numpy as np
+from numpy import array
+import pandas as pd
+import matplotlib.pyplot as plt
+get_ipython().run_line_magic('matplotlib', 'inline')
+from time import time
+
+# Importing libraries for image manipulation, deep-learning and pickling
+from PIL import Image, ImageOps
+from pickle import dump, load
+import tensorflow as tf
+
+# Importing functionalities from 'keras' library
+from keras.preprocessing import sequence
+from keras.models import Sequential
+from keras.layers import LSTM, Embedding, TimeDistributed, Dense, RepeatVector, Activation, Flatten, Reshape, concatenate, Dropout, BatchNormalization
+from keras.optimizers import Adam, RMSprop
+from keras.layers.wrappers import Bidirectional
+from keras.layers.merge import add
+from keras.applications.inception_v3 import InceptionV3
+from keras.preprocessing import image
+from keras.models import Model
+from keras import Input, layers
+from keras import optimizers
+from keras.applications.inception_v3 import preprocess_input
+from keras.preprocessing.text import Tokenizer
+from keras.preprocessing.sequence import pad_sequences
+from keras.utils import to_categorical
+
+#%%
+
+# Importing custom modules
+
+from utils import image_processing
+from utils import caption_writer
+from training_modules import description_processor
+from training_modules import description_properties
+from training_modules import training_functions
+
+# %%
+
+# Loading training descriptions into 'descriptions' dictionary
+
+descriptions= description_processor.load_descriptions()
+
+# %%
+
+# Printing Sample-keys and Sample-descriptions
+
+print("Sample keys : ", list(descriptions.keys())[:5], "\n")
+print("Sample Description 1 : ", descriptions['1000268201'], "\n")
+print("Sample Description 2 : ", descriptions['1000344755'], "\n")
+
+# %%
+
+# Cleaning the description strings making it compatible for training model
+
+description_processor.clean_descriptions(descriptions)
+
+# %%
+
+#Printing cleaned description samples
+
+print("Sample 1 : ", descriptions['1000344755'], "\n")
+print("Sample 2 : ", descriptions['1000268201'], "\n")
+    
+# %%
+
+# Adding all distinct words in the description dictionary to the 'all_desc' set
+
+all_desc=set()
+for key in descriptions.keys():
+    #print(key)
+    for d in descriptions[key]:
+        [all_desc.update(d.split())]
+
+# %%
+
+# Printing the number of distinct words present in the descriptions
+        
+vocabulary=all_desc
+print("size of the vocabulary=",len(vocabulary))
+
+# %%
+
+# Loading and printing the size of the description dataset
+
+filename='results.csv'
+train=description_processor.load_set(filename)
+print('dataset=',len(train))
+
+# %%
+
+# Initializing the images-dataset directory and adding the name of each image into 'img' list
+
```

Comment:
In this part you are just encoding and picking the images which should be done in other file and only the encoded_train_images.pkl should be called here.